src/clj/rems/api/user_settings.clj

```diff
@@ -3,7 +3,19 @@
             [rems.api.schema :as schema]
             [rems.db.user-settings :as user-settings]
             [rems.util :refer [getx-user-id get-user-id]]
-            [ring.util.http-response :refer :all]))
+            [ring.util.http-response :refer :all]
+            [schema.core :as s])
+  (:import (org.joda.time DateTime)))
+
+(s/defschema GetUserSettings
+  {:language s/Keyword
+   :notification-email (s/maybe s/Str)
+   (s/optional-key :ega) {:api-key-expiration-date DateTime}})
+
+(s/defschema UpdateUserSettings
+  {(s/optional-key :language) s/Keyword
+   (s/optional-key :notification-email) (s/maybe s/Str)
+   (s/optional-key :ega) {:api-key-expiration-date DateTime}})
```

Comment:
So what's the use case for setting the expiration date via the update API? I thought the expiration date would be derived from the key, and thus read-only. Or is this just a stepping stone on the way to storing the API key?

Reply:
Good point. It can actually be removed since it is set only from the back-end. Just need to make sure it's available for setting outside the API.

Reply:
If we remove it then we don't need to check for `:enable-ega` here...